pkg/fission-cli/cmd/environment/update.go

```diff
@@ -78,7 +79,20 @@ func (opts *UpdateSubCommand) complete(input cli.Input) (err error) {
 }
 
 func (opts *UpdateSubCommand) run(input cli.Input) error {
+	m := opts.env.ObjectMeta
+	if input.Bool(flagkey.SpecSave) {
+		err := opts.env.Validate()
+		if err != nil {
+			return fv1.AggregateValidationErrors("Environment", err)
+		}
 
+		specFile := fmt.Sprintf("env-%v.yaml", m.Name)
+		err = spec.SpecSaveUpdate(*opts.env, specFile)
+		if err != nil {
+			return errors.Wrap(err, "error saving environment spec")
+		}
+		return nil
```

Comment:
@shubham-bansal96 This will also resolve this issue of only updating the resource spec and not updating the resource on the server, when `fission update % --spec` is used.

WDYT?

Reply:
yes, Lets make the same changes everywhere.
If user provide --spec command while updating function or any other resources, then it should be update the spec file. No server side changes.
In case of spec file, server side changes should be made only using command fission spec apply.